index/scorch/segment/zap/docvalues.go

```diff
@@ -254,20 +271,22 @@ func (s *SegmentBase) VisitDocumentFieldTerms(localDocNum uint64, fields []strin
 
 		if dvIter, exists := s.fieldDvReaders[fieldIDPlus1-1]; exists &&
 			dvIter != nil {
-			dvIterClone = dvIter.cloneInto(dvIterClone)
+			if _, ok := dvs.dvrs[fieldIDPlus1-1]; !ok {
```

Comment:
looks fieldIDPlus1-1 is recomputed a few times?  might as well do it once with a helper var?

Reply:
also, there's a lot of map lookups (dvs.dvrs[fieldIDPlus1-1]) -- ought to be able to do it once and reuse?